core/src/main/java/org/mifos/mobilewallet/core/domain/usecase/standinginstruction/CreateStandingTransaction.kt

```diff
@@ -0,0 +1,191 @@
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(object : Subscriber<ClientAccounts>() {
+                    override fun onCompleted() {
+
+                    }
+
+                    override fun onError(e: Throwable) {
+                        useCaseCallback.onError(Constants.ERROR_FETCHING_FROM_ACCOUNT)
+                    }
+
+                    override fun onNext(clientAccounts: ClientAccounts) {
+                        val accounts = clientAccounts.savingsAccounts
+                        if (accounts != null && accounts.size != 0) {
+                            var walletAccount: SavingAccount? = null
+                            for (account in accounts) {
+                                if (account.productId ==
+                                        Constants.WALLET_ACCOUNT_SAVINGS_PRODUCT_ID) {
+                                    walletAccount = account
+                                    break
+                                }
+                            }
+                            walletAccount?.let{
+                                fromAccount = walletAccount
+                                fetchToAccountDetails()
+                            }?: useCaseCallback.onError(Constants.NO_WALLET_FOUND)
+
+                        } else {
+                            useCaseCallback.onError(Constants.ERROR_FETCHING_FROM_ACCOUNT)
+                        }
+                    }
+                })
+    }
+
+    private fun fetchToAccountDetails() {
+        apiRepository.getAccounts(requestValues.toClientId)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(object : Subscriber<ClientAccounts>() {
+                    override fun onCompleted() {}
+                    override fun onError(e: Throwable) {
+                        useCaseCallback.onError(Constants.ERROR_FETCHING_TO_ACCOUNT)
+                    }
+
+                    override fun onNext(clientAccounts: ClientAccounts) {
+                        val accounts = clientAccounts.savingsAccounts
+                        if (accounts != null && accounts.size != 0) {
+                            var walletAccount: SavingAccount? = null
+                            for (account in accounts) {
+                                if (account.productId ==
+                                        Constants.WALLET_ACCOUNT_SAVINGS_PRODUCT_ID) {
+                                    walletAccount = account
+                                    break
+                                }
+                            }
+                            walletAccount?.let{
+                                toAccount = walletAccount
+                                createNewStandingInstruction()
+                            }?: useCaseCallback.onError(Constants.NO_WALLET_FOUND)
```

Comment:
`Constants.NO_WALLET_FOUND` do we show this error message in UI? If yes then it would be much better to add inside `string.xml` file

Reply:
Thanks for pointing this out